components/dashboard/src/App.tsx

```diff
@@ -119,6 +119,28 @@ function isWebsiteSlug(pathName: string) {
     return slugs.some((slug) => pathName.startsWith("/" + slug + "/") || pathName === "/" + slug);
 }
 
+// A wrapper for <Route> that redirects to the workspaces screen if the user isn't a admin.
+// This wrapper only accepts the component property
+function AdminRoute({ component }: any) {
+    const { user } = useContext(UserContext);
+    return (
+        <Route
+            render={({ location }: any) =>
+                user?.rolesOrPermissions?.includes("admin") ? (
+                    <Route component={component}></Route>
```

Comment:
🤔 the nested `Route` seems suspicious: shouldn't this just be `{component}`?

Look at how they use `{children}` in the original example:

```tsx
function PrivateRoute({ children, ...rest }) {
  let auth = useAuth();
  return (
    <Route
      {...rest}
      render={({ location }) =>
        auth.user ? (
          children
        ) : (
          <Redirect
            to={{
              pathname: "/login",
              state: { from: location }
            }}
          />
        )
      }
    />
  );
```

Reply:
Nope, That doesn't work! Looks like we need to pass along with the `Route` tag. It works in the example for children because its some HTML code there but here we need this properties of the current tag i.e `component`

Reply:
It *should* work... 🤔 maybe I'll take it for a spin some time in the future and see what happens 🤷